userspace/plugin/plugin_api.h

```diff
@@ -970,13 +1031,40 @@ typedef struct
 	//
 	// Return an updated set of metrics provided by this plugin.
 	// Required: no
+	// Arguments:
 	// - s: the plugin state, returned by init(). Can be NULL.
 	// - num_metrics: lenght of the returned metrics array.
 	//
 	// Return value: Pointer to the first element of the metrics array.
 	// 'num_metrics' must be set to the lenght of the array before returning
 	// and it can be set to 0 if no metrics are provided.
 	ss_plugin_metric* (*get_metrics)(ss_plugin_t* s, uint32_t* num_metrics);
+
+	//Capture listening capability API
+	struct
+	{
+		//
+		// Called by the framework when the event capture opens.
+		//
+		// Required: no
+		// Arguments:
+		// - s: the plugin state, returned by init(). Can be NULL.
+		// - i: input containing vtables for performing table operations and subscribe/unsubscribe async routines
+		//
+		// Return value: A ss_plugin_rc with values SS_PLUGIN_SUCCESS or SS_PLUGIN_FAILURE.
+		ss_plugin_rc (*capture_open)(ss_plugin_t* s, const ss_plugin_capture_listen_input* i);
+
+		//
+		// Called by the framework when the event capture closes.
+		//
+		// Required: yes if capture_open is defined
+		// Arguments:
+		// - s: the plugin state, returned by init(). Can be NULL.
```

Comment:
Same thing. I assume this is a copy-paste thing? But also, what do we do if ->capture_close returns an error? Is the return value meaningful here?

Reply:
> what do we do if ->capture_close returns an error? Is the return value meaningful here?

Currently the return value of `capture_open`/`capture_close` is ignored by the inspector.
But in the future we may need to know if something went wrong in the plugin during `capture_open`/`capture_close`, we will be able to do that without introducing a breaking change in the plugin api.

Reply:
Yes, but what are you going to do about it? It's like a destructor returning an error. I don't think there's any way to handle an error here, except for feeling a little bit sad and moving on (you're not going to cancel the capture stop anyway).

Still, not going to protest too hard about it ;) it just seems useless, not actively harmful

(an error from `capture_open` should presumably tear down the whole inspector though, or maybe disable the plugin or something; in any case, it's definitely actionable and should be reported)